src/ucs/datastruct/ptr_array.c

```diff
@@ -202,12 +203,52 @@ unsigned ucs_ptr_array_insert(ucs_ptr_array_t *ptr_array, void *value)
     return element_index;
 }
 
+unsigned ucs_ptr_array_alloc(ucs_ptr_array_t *ptr_array, unsigned element_count)
+{
+    unsigned free_iter, new_size, element_index;
+
+    if (element_count == 0) {
+        return 0;
+    }
+
+    free_iter     = 1;
```

Comment:
why starting from 1 and not 0?

Reply:
saves one loop iteration guaranteed because an element from the free-list is always vacant. Looked more readable than checking if we've reached `element_count-1` empty cells. I'll add a comment :)